src/main/java/com/github/princesslana/eriscasper/ErisCasper.java

```diff
@@ -42,14 +42,16 @@
   private final Routes routes;
   private final Optional<ShardPayload> shard;
 
+  private Gateway gateway;
+
   private ErisCasper(BotToken token, Optional<ShardPayload> shard) {
     this.token = token;
     this.shard = shard;
     routes = new Routes(token, httpClient, jackson);
   }
 
   private Observable<Event> getEvents() {
-    Gateway gateway = Gateway.create(httpClient, payloads);
+    this.gateway = Gateway.create(httpClient, payloads);
```

Comment:
Why can't we create this in the constructor?

Reply:
2 different methods, we could merge the methods but that would break the separation we want to maintain.

Reply:
Oh wait, it's merged into 1, didn't notice.

Reply:
Yeah, they come into 2 separate methods.
https://hastebin.com/rorokiluxu.java
If we merged them it would ruin the flow of the bot.

Reply:
We could hold the botcontext as a field in the ErisCasper instance but it might not be worth it to do that over the gateway.

Reply:
If you're referring to the ErisCasper constructor we could do that and finalize the gateway variable. Didn't really understand what you were going at until just now.

Reply:
Yeah, in the ErisCasper constructor. With final and then we know it's always not null too 👍